javaslang/src/test/java/javaslang/control/TryTest.java

```diff
@@ -912,9 +911,9 @@ private RuntimeException error() {
 
     private static <T, X extends Throwable> Try<T> failure(Class<X> exceptionType) {
         try {
-            final X exception = exceptionType.newInstance();
+            final X exception = exceptionType.getConstructor().newInstance();
             return Try.failure(exception);
-        } catch (InstantiationException | IllegalAccessException x) {
+        } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException ignored) {
```

Comment:
If you wanted to be cleaner, you could also catch `ReflectiveOperationException` (see [javadoc](https://docs.oracle.com/javase/7/docs/api/java/lang/ReflectiveOperationException.html)) which is the new superclass for all reflection based exceptions which came in with Java 1.7.